src/Common/Misc/BusyMonitor/index.js

```diff
@@ -39,9 +39,8 @@ function busyMonitor(publicAPI, model) {
   };
 
   publicAPI.busy = (promise) => {
-    promise.then(success, error);
     checkNotifyStatus(1);
-    return promise;
+    return promise.then(success, error);
```

Comment:
Sorry misread what `checkNotifyStatus` was supposed to do. Yes that should work.